src/vs/workbench/parts/extensions/electron-browser/extensionTipsService.ts

```diff
@@ -234,27 +253,93 @@ export class ExtensionTipsService extends Disposable implements IExtensionTipsSe
 						badRecommendationsString
 					);
 				}
-
 				return validRecommendations;
 			});
 		}
-
 		return TPromise.as([]);
+	}
+
+	private processWorkspaceIgnores(extensionsContent: IExtensionsConfigContent): string[] {
+		const regEx = new RegExp(EXTENSION_IDENTIFIER_PATTERN);
+
+		let countBadIDs = 0;
+		let badIDsString = '';
+		let cleansedIDs = extensionsContent.unwantedRecommendations.filter(element => {
+			if (!regEx.test(element)) {
+				countBadIDs++;
+				badIDsString += `${element} (bad format) Expected: <provider>.<name>\n`;
+				return false;
+			}
+			return true;
+		});
 
+		if (countBadIDs > 0 && this.notificationService) {
+			this.notificationService.warn(
+				'The below ' +
+				countBadIDs +
+				' extension(s) in workspace ignored recommendations have formatting issues:\n' +
+				badIDsString
+			);
+		}
+		return cleansedIDs;
+	}
+
+	refreshAllIgnoredRecommendations(): TPromise<void> {
+		const globallyIgnored = Promise.resolve(<string[]>JSON.parse(this.storageService.get('extensionsAssistant/ignored_recommendations', StorageScope.GLOBAL, '[]')));
+		const workspaceIgnored = this.getWorkspaceIgnores();
+		return TPromise.join([globallyIgnored, workspaceIgnored]).then(ignored => {
+			this._globallyIgnoredRecommendations = distinct(ignored[0]).map(id => id.toLowerCase());
+			this._workspaceIgnoredRecommendations = distinct(ignored[1]).map(id => id.toLowerCase());
+			this._allIgnoredRecommendations = distinct([...this._globallyIgnoredRecommendations, ...this._workspaceIgnoredRecommendations]);
+
+			this._allWorkspaceRecommendedExtensions = this._allWorkspaceRecommendedExtensions.filter((id) => this.isExtensionAllowedToBeRecommended(id));
+			this._dynamicWorkspaceRecommendations = this._dynamicWorkspaceRecommendations.filter((id) => this.isExtensionAllowedToBeRecommended(id));
+
+			let filteredFileBased: { [id: string]: number; } = {};
+			forEach(this._fileBasedRecommendations, entry => {
+				if (this.isExtensionAllowedToBeRecommended(entry.key)) {
+					filteredFileBased[entry.key] = this._fileBasedRecommendations[entry.key];
+				}
+			});
+
+			let filteredExeBased: { [id: string]: string; } = {};
+			forEach(this._exeBasedRecommendations, entry => {
+				if (this.isExtensionAllowedToBeRecommended(entry.key)) {
+					filteredExeBased[entry.key] = this._exeBasedRecommendations[entry.key];
+				}
+			});
+
+			this._exeBasedRecommendations = filteredExeBased;
+			this._fileBasedRecommendations = filteredFileBased;
+		});
+	}
+
+	getAllIgnoredRecommendations(): IIgnoredRecommendations {
+		return {
+			workspace: this._workspaceIgnoredRecommendations,
+			global: this._globallyIgnoredRecommendations
+		};
+	}
+
+	private isExtensionAllowedToBeRecommended(id: string): boolean {
+		return this._allIgnoredRecommendations.indexOf(id.toLowerCase()) === -1;
```

Comment:
This seems to be the only place where `_allIgnoredRecommendations` is used. We can get rid of the extra variable and use `_workspaceIgnoredRecommendations`  and `_globallyIgnoredRecommendations` here directly right?